include/libdnf/rpm/package_query.hpp

```diff
@@ -77,285 +77,495 @@ class PackageQuery : public PackageSet {
+    PackageQuery & filter_evr(
+        const std::vector<std::string> & patterns, libdnf::sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
 
-    /// cmp_type could be only libdnf::sack::QueryCmp::EQ, NEQ, GLOB, NOT_GLOB.
-    PackageQuery & filter_epoch(
+    /// Filter packages by their `name-[epoch:]version-release.arch`. The following matches are tolerant to omited 0 epoch: `EQ`, `NEQ`, `GT`, `GTE`, `LT`, `LTE`.
+    ///
+    /// @param patterns         A vector of strings the filter is matched against.
+    /// @param cmp              A comparison (match) operator, defaults to `QueryCmp::EQ`.
+    ///                         Supported values: `EQ`, `NEQ`, `GT`, `GTE`, `LT`, `LTE`, `GLOB`, `NOT_GLOB`, `IGLOB`, `NOT_IGLOB`, `IEXACT`, `NOT_IEXACT`.
+    /// @since 5.0
+    //
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char *match) - cmp_type = HY_PKG_NEVRA
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char **matches) - cmp_type = HY_PKG_NEVRA
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char *match) - cmp_type = HY_PKG_NEVRA_STRICT
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char **matches) - cmp_type = HY_PKG_NEVRA_STRICT
+    PackageQuery & filter_nevra(
         const std::vector<std::string> & patterns, libdnf::sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
 
-    /// cmp_type could be only libdnf::sack::QueryCmp::EQ, NEQ, GLOB, NOT_GLOB, IEXACT, NOT_IEXACT, ICONTAINS, NOT_ICONTAINS, IGLOB, NOT_IGLOB, CONTAINS, NOT_CONTAINS.
+    /// Filter packages by their `name-[epoch:]version-release.arch` based on values set in the `nevra` object.
     ///
-    /// @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char *match) - cmp_type = HY_PKG_FILE
-    /// @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char **matches) - cmp_type = HY_PKG_FILE
-    PackageQuery & filter_file(
-        const std::vector<std::string> & patterns, libdnf::sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+    /// @param nevra            A Nevra object the filter is matched against.
+    /// @param cmp              A comparison (match) operator, defaults to `QueryCmp::EQ`.
+    ///                         Supported values: `EQ`, `NEQ`, `GLOB`, `NOT_GLOB`, `IEXACT`, `NOT_IEXACT`, `IGLOB`, `NOT_IGLOB`.
+    /// @since 5.0
+    PackageQuery & filter_nevra(
+        const libdnf::rpm::Nevra & nevra, libdnf::sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
 
-    /// cmp_type could be only libdnf::sack::QueryCmp::EQ, NEQ, GLOB, NOT_GLOB, IEXACT, NOT_IEXACT, ICONTAINS, NOT_ICONTAINS, IGLOB, NOT_IGLOB, CONTAINS, NOT_CONTAINS.
+    /// Filter packages by their `name-[epoch:]version-release.arch` attributes of the packages in the `package_set`.
```

Comment:
What about to replace `name-[epoch:]version-release.arch` by `name`, `epoch`, `version`, `release`, and `arch` attributes?

Reply:
I intentionally used this to indicate what's the input format.
Individual values do not describe the input format at all.